src/main/java/app/ProgramRepository.java

```diff
@@ -12,4 +12,5 @@
 
     List<Program> findAll();
     List<Program> findByName(String name);
+    List<Program> findById(Long id);
```

Comment:
Can't you just have it find A program instead of a List (which it really would never get because unique IDs, it would always be a list of size 1). This would have you not have to get the 0th index element of a list of just one parameter too.
Or is there a reason this was done ?

Reply:
You're right, I'll make the change.